test/backend/server.test.tsx

```
@@ -0,0 +1,205 @@
+          target: 'Kewl kids use Haskell',
+          version: 'v1.1.1',
+          options: 'options should be an object and not a string',
+          buildStatus: BuildStatus.NOTRUN,
+        };
+
+        const createdPlan = await app.service('plans').create(planData);
+
+        const sdkData = { planId: createdPlan.id };
+
+        const spy = jest.spyOn(dependencyToMock, 'generateSdk').mockImplementation(() => {
+          const swaggerCodegenMalformedOptionsResponse = {
+            code: 500,
+            type: 'unknown',
+            message: 'something bad happened',
+          };
+          throw new Error(swaggerCodegenMalformedOptionsResponse.message);
+        });
+
+        let createdSdk;
+        let errorMessage;
+        try {
+          createdSdk = await app.service('sdks').create(sdkData);
+        } catch (err) {
+          errorMessage = err.toString();
+        }
+
+        // generateSdk() called once.
+        expect(spy).toHaveBeenCalledTimes(1);
+        // SDK not created.
+        expect(createdSdk).toBe(undefined);
+        // Error throw with right message.
+        expect(errorMessage).toEqual('Error: something bad happened');
+      });
+
+      test('create a sdk error, invalid path', async () => {
+        const specData: Spec = {
+          title: 'Dummy specification title',
+          description: 'A description of my specification',
+          path: 'this fake path will lead to an error this time yay',
+        };
+        const createdSpec = await app.service('specifications').create(specData);
+
+        const planData: Plan = {
+          specId: createdSpec.id,
+          target: 'Kewl kids use Haskell',
+          version: 'v1.1.1',
+          buildStatus: BuildStatus.NOTRUN,
+        };
+
+        const createdPlan = await app.service('plans').create(planData);
+
+        const sdkData = { planId: createdPlan.id };
+        const spy = jest.spyOn(dependencyToMock, 'generateSdk').mockImplementation(() => {
```

I think its better to create a `__mocks__` mock for this. Lots of tests are going to need to mock out `generateSdk`.

---

I assume __mocks__ is just a file of dummy data for the response? If so can you change the response cuz the other tests for failure would return different stuff

---

[Read this](https://jestjs.io/docs/en/manual-mocks)
It's where you can store mocked modules so when you call `jest.mock('module-path-thing')` it uses the mock in the __mocks__ folder.

---

I gave it a shot but fked something up, following mocking es6 classes it gets to 
```
const plan = await plans.get(context.data.planId);
const spec = await specs.get(plan.specId);
const generationResponse = await generateSdk(logger, spec, plan);
```
and errors saying `generateSdk()` is not a function.

I think since we are importing `generateSdk()` in `server.tsx` and calling it from there testing gets a bit hacky (like what I needed to do with import * as.. then mocking that out). Will come back to it later but don't think it is a big deal atm.